plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/BaseJdbcTableStatisticsTest.java

```diff
@@ -0,0 +1,354 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.Session;
+import io.trino.SystemSessionProperties;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.sql.TestTable;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.util.Locale;
+
+import static com.google.common.base.Throwables.getStackTraceAsString;
+import static io.trino.testing.sql.TestTable.randomTableSuffix;
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public abstract class BaseJdbcTableStatisticsTest
+        extends AbstractTestQueryFramework
+{
+    // Currently this class serves as a common "interface" to define cases that should be covered.
+    //  TODO extend it to provide reusable blocks to reduce boiler-plate.
+
+    @BeforeClass
+    public void setUpTables()
+    {
+        setUpTableFromTpch("region");
+        setUpTableFromTpch("nation");
+    }
+
+    private void setUpTableFromTpch(String tableName)
```

Comment:
> We just need to do gatherStats on them here.

that's the reason why it's done differently

if we do this separately, we may end up with some tables not analyzed.

Anyway, i don't want to introduce any optional changes here. There is more stuff Starburst gonna open source, and it would be beneficial to everyone if optional improvements are done afterwards. That simply means the super cool value is delivered sooner to all the users.